### PR TITLE
Add missing locations in the parsetree

### PR DESCRIPTION
**Pull request by lefessan**
They are not propagated to the typedtree yet.

### CODE REVIEW

**Comment by lefessan** [comment]
This PR compiles fine on Travis, but then fail because `Camlp4` is not compatible with the changes to the parsetree. Compiling `Camlp4` in Travis is really weird, as any modification to the Parsetree is bound to make the checks fail.

File: otherlibs/num/Makefile [hunk 1]
```diff
@@ -15,8 +15,4 @@
 
 include Makefile.shared
 
-depend:
```
Comment (alainfrisch) [inline otherlibs/num/Makefile:18]: This chunk seems unrelated to this PR.

**Comment by alainfrisch** [comment]
I'm in favor of the change, although many of the interesting use cases I can see for the new locations would involve having them in the Typedtree as well (to be used through .cmt files).

Even with the current changes to the Parsetree only, this could improve the location for some error messages (although this is not implemented in the PR, AFAICT).

**Comment by alainfrisch** [comment]
@lefessan Do you plan to add the locations to Typedtree as well?

What about my note about a chunk unrelated to the PR?
